src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -32,6 +33,9 @@ public class BlockValidationContext
 
         /// <summary>If the block validation failed this will be set with the reason of failure.</summary>
         public ConsensusError Error { get; set; }
+
+        /// <summary>Whether to skip block validation for this block due to either a checkpoint or assumevalid hash set.</summary>
+        public bool SkipValidation { get; set; }
```

Comment:
This should go in the other context (this context is wrongly named its more about downloading blocks)

Reply:
you sure? i would put it exactly here as well. i think this is the correct context, there is that error that you receive if validation failed ...

Reply:
If this is for only downloading blocks we need to change the comments too as they all refer to "downloaded or mined".

Reply:
To me this context is for interacting with the consensus (you send  context and check the error), but stuff that are part of the consensus validation go in the other context.

Well let me think about it again, I will review the code better tomorrow.